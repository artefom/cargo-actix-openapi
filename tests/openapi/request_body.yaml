openapi: "3.0.3"
info:
  version: "1.0"
  title: "Hello World API"
paths:
  /hello/{user}:
    parameters:
      - name: user
        in: path
        required: true
        description: The name of the user to greet.
        schema:
          type: string
    post:
      operationId: greet_user
      summary: Returns a greeting to the user!
      description: Some detailed description
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: string
      responses:
        200:
          description: Returns the greeting.
          content:
            application/json:
              schema:
                type: string
