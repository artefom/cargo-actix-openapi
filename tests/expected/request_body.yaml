api:
  definitions:
  - name: GreetUserPath
    data: !Struct
      doc: null
      properties:
      - name: user
        rename: user
        default: null
        ptype: String
  operations:
  - name: greet_user
    path: /hello/{user}
    method: Post
    doc: Returns a greeting to the user!
    param_path: web::Path<GreetUserPath>
    param_query: null
    param_body: web::Json<String>
    response: web::Json<String>
